MAUI/DataGrid/columns.md

```diff
@@ -0,0 +1,205 @@
+
+{% tabs %}
+{% highlight c# %}
+private void SfDataGrid_AutoGeneratingColumn(object sender, DataGridAutoGeneratingColumnEventArgs e)
+{
+    if (e.Column.MappingName == "Freight")
+    {
+        e.Column.Format = "C";
+    }
+    else if (e.Column.MappingName == "ShippingDate")
+    {
+        e.Column.Format = "MMMM dd";
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Manually generate columns
+
+The `SfDataGrid` allows to define the columns manually by adding the [DataGridColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html) objects to the [SfDataGrid.Columns](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_Columns) collection. If you want to show only the manually defined columns in the view, you can achieve it by setting the `SfDataGrid.AutoGenerateColumnsMode` property to `None`.
+
+There are different types of columns available. Any column can be created based on the requirements from both XAML and code.
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid" AutoGenerateColumnsMode="None" ItemsSource="{Binding OrderInfoCollection}" >
+    <syncfusion:SfDataGrid.Columns>
+        <syncfusion:DataGridNumericColumn HeaderText="Order ID" MappingName="OrderID"/>
+        <syncfusion:DataGridTextColumn  HeaderText="Customer ID" MappingName="CustomerID"/>
+        <syncfusion:DataGridTextColumn  HeaderText="Customer" MappingName="Customer"/>
+        <syncfusion:DataGridTextColumn  HeaderText="Ship City" MappingName="ShipCity"/>
+        <syncfusion:DataGridTextColumn  HeaderText="Ship Country" MappingName="ShipCountry"/>
+    </syncfusion:SfDataGrid.Columns>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+this.sfDataGrid.AutoGenerateColumnsMode = AutoGenerateColumnsMode.None;
+
+DataGridNumericColumn orderIdColumn = new DataGridNumericColumn { HeaderText = "Order ID", MappingName = "OrderID" };
+DataGridTextColumn customerIdColumn = new DataGridTextColumn { HeaderText = "Customer ID", MappingName = "CustomerID" };
+DataGridTextColumn customerColumn = new DataGridTextColumn { HeaderText = "Customer", MappingName = "Customer" };
+DataGridTextColumn shipCityColumn = new DataGridTextColumn { HeaderText = "Ship City", MappingName = "ShipCity" };
+DataGridTextColumn shipCountryColumn = new DataGridTextColumn { HeaderText = "Ship Country", MappingName = "ShipCountry" };
+
+this.sfDataGrid.Columns.Add(orderIdColumn);
+this.sfDataGrid.Columns.Add(customerIdColumn);
+this.sfDataGrid.Columns.Add(customerColumn);
+this.sfDataGrid.Columns.Add(shipCityColumn);
+this.sfDataGrid.Columns.Add(shipCountryColumn);
+{% endhighlight %}
+{% endtabs %}
+
+## Column manipulation
+
+You can get the columns from the `SfDataGrid.Columns` property.
+
+### Adding column to DataGrid
+
+You can add a column to the DataGrid at runtime by adding instance of a `DataGridColumn` to the `SfDataGrid.Columns` collection.

```

Comment:
You can add a column to the DataGrid at runtime by adding an instance of a `DataGridColumn` to the `SfDataGrid.Columns` collection.
 You can add a column to the DataGrid at runtime by adding an instance of a `DataGridColumn` to the `SfDataGrid.Columns` collection.

Reply:
Modified.